### PR TITLE
[build] Add version script and update Mapbox project targets (only)

### PR DESCRIPTION
**Pull request by julianrex**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement).

## Pull request checklist:
 - [X] Briefly describe the changes in this PR.
 - [X] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'

### Summary of changes

This PR updates the version numbers for the Mapbox xcode project targets, using a script modified from `mapbox-gl-native-ios`. This does not yet include the Examples, Debug App or StressTest projects.

### User impact (optional)

Generated frameworks will now include the version number.

### CODE REVIEW

File: scripts/release/update-version.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+
```
Comment (neelmistry94) [inline scripts/release/update-version.sh:2]: add a usage comment at the top

File: scripts/release/update-version.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+SEM_VERSION=$1
+SEM_VERSION=${SEM_VERSION/#v}
+SHORT_VERSION=${SEM_VERSION%-*}
+
+step "Version ${SEM_VERSION}"
+
+step "Updating CocoaPods podspecs to version ${SEM_VERSION}…"
+
+find . -type f -name '*.podspec' -exec sed -i '' "s/^ *version *=.*$/  version = '${SEM_VERSION}'/" {} +
```
Comment (neelmistry94) [inline scripts/release/update-version.sh:24]: why are we updating the podspec here?

Comment (julianrex) [reply scripts/release/update-version.sh:24]: This is a port from <10.0. This is likely to clash with your PR 😄

**Review by neelmistry94 — APPROVED** [review]

**Comment by julianrex** [comment]
Noting that changes requested above will be integrated in #83

**Pull request merged**